app/obolapi/api.go

```diff
@@ -15,46 +16,68 @@ import (
 	"github.com/obolnetwork/charon/cluster"
 )
 
+const (
+	// launchpadReturnPath is the URL path at which one can find details for a given cluster lock hash.
+	launchpadReturnPath = "/clusters/details"
+
+	// lockQueryStr is the URL query string for the lock hash parameter.
+	lockQueryStr = "lockHash"
+)
+
 // New returns a new Client.
-func New(url string) Client {
-	return Client{
-		baseURL: url,
+func New(urlStr string) (Client, error) {
+	apiURL, err := url.ParseRequestURI(urlStr)
+	if err != nil {
+		return Client{}, errors.Wrap(err, "could not parse Obol API URL")
 	}
+
+	return Client{
+		baseURL: apiURL,
+	}, nil
 }
 
 // Client is the REST client for obol-api requests.
 type Client struct {
-	baseURL string // Base obol-api URL
+	baseURL *url.URL // Base obol-api URL
 }
 
 // PublishLock posts the lockfile to obol-api.
 func (c Client) PublishLock(ctx context.Context, lock cluster.Lock) error {
 	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
 	defer cancel()
 
-	addr, err := url.JoinPath(c.baseURL, "lock")
-	if err != nil {
-		return errors.Wrap(err, "invalid address")
-	}
-
-	url, err := url.Parse(addr)
-	if err != nil {
-		return errors.Wrap(err, "invalid endpoint")
-	}
+	addr := *c.baseURL
```

Comment:
I would keep baseURL a string since this pointer trick is easy to miss when adding more functions which can result in bugs.

Reply:
What do you think about this? Validate url in `New()`, and provide a `url()` method that returns it as `*url.URL`, panic-ing if parsing fails: since validation happens in `New()`, if it fails later it's a very weird situation:

```go
// New returns a new Client.
func New(urlStr string) (Client, error) {
	_, err := url.ParseRequestURI(urlStr) // check that urlStr is valid
	if err != nil {
		return Client{}, errors.Wrap(err, "could not parse Obol API URL")
	}

	return Client{
		baseURL: urlStr,
	}, nil
}

// Client is the REST client for obol-api requests.
type Client struct {
	baseURL string // Base obol-api URL
}

func (c Client) url() *url.URL {
	baseURL, err := url.ParseRequestURI(c.baseURL)
	if err != nil {
		panic(errors.Wrap(err, "could not parse Obol API URL, this should never happen"))
	}

	return baseURL
}
```

Reply:
sure, could also just return the error from `url()`, don't think the effort to omit errors is super valuable.